corehq/apps/export/models/new.py

```diff
@@ -1233,27 +1240,103 @@ def get_latest_export_schema(domain, app_id, form_xmlns):
 
     @staticmethod
     def _process_app_build(current_schema, app, form_xmlns):
-        xform = app.get_form_by_xmlns(form_xmlns, log_missing=False)
-        if not xform:
+        form = app.get_form_by_xmlns(form_xmlns, log_missing=False)
+        if not form:
             return current_schema
-        case_updates = xform.get_case_updates(xform.get_module().case_type)
-        xform = xform.wrapped_xform()
+
+        case_updates = form.get_case_updates(form.get_module().case_type)
+        xform = form.wrapped_xform()
+        repeats_with_subcases = {
+            subcase_action for subcase_action in form.actions.subcases
+            if subcase_action.repeat_context
+        }
         xform_schema = FormExportDataSchema._generate_schema_from_xform(
             xform,
             case_updates,
             app.langs,
             app.copy_of or app._id,  # If it's not a copy, must be current
             app.version,
         )
-        return FormExportDataSchema._merge_schemas(current_schema, xform_schema)
+        schemas = [current_schema, xform_schema]
+        if repeats_with_subcases:
+            repeat_case_schema = FormExportDataSchema._generate_schema_from_repeat_subcases(
+                xform,
+                repeats_with_subcases,
+                app.langs,
+                app.copy_of or app._id,  # If it's not a copy, must be current
+                app.version,
+            )
+            schemas.append(repeat_case_schema)
+
+        return FormExportDataSchema._merge_schemas(*schemas)
+
+    @staticmethod
+    def _generate_schema_from_repeat_subcases(xform, repeats_with_subcases, langs, app_id, app_version):
+        """
+        This generates a FormExportDataSchema for repeat groups that generate subcases.
+
+        :param xform: An XForm instance
+        :param repeats_with_subcases: A list of OpenSubCaseAction classes that have a
+            repeat_context.
+        :param langs: An array of application languages
+        :param app_id: The app_id of the corresponding app
+        :param app_version: The build number of the app
+        :returns: An instance of a FormExportDataSchema
+        """
+
+        repeats = FormExportDataSchema._get_repeat_paths(xform, langs)
+        schema = FormExportDataSchema()
```

Comment:
minor: if this was a `classmethod` instead of a `staticmethod` you could `cls._get_repeat_paths(...)` and `cls()` here. This could be applied to a bunch of other static methods around here as well.